test/hooks.on-listen.test.js

```diff
@@ -0,0 +1,679 @@
+'use strict'
+
+const { t, before } = require('tap')
+const Fastify = require('../fastify')
+const fp = require('fastify-plugin')
+const dns = require('dns').promises
+
+let localhost
+
+before(async function () {
+  const lookup = await dns.lookup('localhost')
+  localhost = lookup.address
+})
+
+t.test('localhost onListen should be called in order', t => {
+  t.plan(2)
+  const fastify = Fastify()
+  t.teardown(fastify.close.bind(fastify))
+  let order = 0
+
+  fastify.addHook('onListen', function (done) {
+    t.equal(order++, 0, '1st called in root')
+    done()
+  })
+
+  fastify.addHook('onListen', function (done) {
+    t.equal(order++, 1, '2nd called in root')
+    done()
+  })
+
+  fastify.listen({
+    host: 'localhost',
+    port: 0
+  })
+})
+
+t.test('localhost async onListen should be called in order', async t => {
+  t.plan(2)
+  const fastify = Fastify()
+  t.teardown(fastify.close.bind(fastify))
+  let order = 0
+
+  fastify.addHook('onListen', async function () {
+    t.equal(order++, 0, '1st async called in root')
+  })
+
+  fastify.addHook('onListen', async function () {
+    t.equal(order++, 1, '2nd async called in root')
+  })
+
+  await fastify.listen({
+    host: 'localhost',
+    port: 0
+  })
+})
+
+t.test('localhost onListen sync should log errors as warnings and continue', t => {
+  t.plan(3)
+  const fastify = Fastify()
+  t.teardown(fastify.close.bind(fastify))
+
+  fastify.addHook('onListen', function (done) {
+    t.pass('called in root')
+    done()
+  })
+
+  fastify.addHook('onListen', function (done) {
+    t.pass('called onListen error')
+    throw new Error('FAIL ON LISTEN2')
+  })
+
+  fastify.addHook('onListen', function (done) {
+    t.pass('onListen hooks continue after error')
+    done()
+  })
+
+  fastify.listen({
+    host: 'localhost',
+    port: 0
+  })
+})
+
+t.test('localhost onListen async should log errors as warnings and continue', async t => {
```

Comment:
If we do not intercept the logs to test out at least the logging function is being called, this test is kinda the same as previous one

Reply:
You can use this one as a reference 👍 
https://github.com/fastify/fastify/blob/e272ce5af0d4b17a736481a73afb66877eb81549/test/close.test.js#L306